gello-world/src/components/login/login.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import { FormErrors } from "./FormErrors.js";
+export class Login extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			username: "",
+			password: "",
+			isChecked: true,
+			formErrors: { username: "", password: "" },
+			usernameValid: false,
+			passwordValid: false,
+			formValid: false
+		};
+	}
+	handleUserInput = e => {
+		const name = e.target.name;
+		const value = e.target.value;
+		this.setState({ [name]: value }, () => {
+			this.validateField(name, value);
+		});
+		this.setState({ value: e.target.username });
+	};
+
+	handleChecked(e) {
+		this.setState({ isChecked: !this.state.isChecked });
+		console.log(this.state.isChecked);
+	}
+	validateField(fieldName, value) {
+		let fieldValidationErrors = this.state.formErrors;
+		let usernameValid = this.state.usernameValid;
+		let passwordValid = this.state.passwordValid;
+
+		switch (fieldName) {
+			case "username":
+				usernameValid = value.match(/^[a-zA-Z0-9]+$/);
+				fieldValidationErrors.username = usernameValid
+					? ""
+					: " используйте только a-z A-Z 0-9";
+
+				break;
+			case "password":
+				passwordValid = value.length >= 6;
+				fieldValidationErrors.password = passwordValid ? "" : " слишком короткий";
+
+				break;
+			default:
+				break;
+		}
+		this.setState(
+			{
+				formErrors: fieldValidationErrors,
+				usernameValid: usernameValid,
+				passwordValid: passwordValid
+			},
+			this.validateForm
+		);
+	}
+
+	validateForm() {
+		this.setState({
+			formValid: this.state.usernameValid && this.state.passwordValid
+		});
+	}
+
+	errorClass(error) {
+		return error.length === 0 ? "" : "has-error";
+	}
+	sing() {
```

Comment:
https://www.merriam-webster.com/dictionary/sing

Reply:
I changed the name of the function to send method)